fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -301,13 +310,42 @@ def self.xcode_list_timeout
     end
 
     # @internal to module
-    # runs the specified command and kills it if timeouts
-    # @raises Timeout::Error if timeout is passed
+    def self.xcode_build_settings_timeout
+      (ENV['FASTLANE_XCODEBUILD_SETTINGS_TIMEOUT'] || 10).to_i
+    end
+
+    # @internal to module
+    def self.xcode_build_settings_retries
+      (ENV['FASTLANE_XCODEBUILD_SETTINGS_RETRIES'] || 3).to_i
+    end
+
+    # @internal to module
+    # runs the specified command and kills it if timeouts, optionally retries before killing
+    # @raises Timeout::Error if timeout is passed after all retry attempts
     # @returns the output
-    # Note: currently affected by fastlane/fastlane_core#102
-    def self.run_command(command, timeout: 0)
+    # Note: - currently affected by https://github.com/fastlane/fastlane/issues/1504
+    #       - retry feature added to solve https://github.com/fastlane/fastlane/issues/4059
+    def self.run_command(command, timeout: 0, retries: 0, print: true)
       require 'timeout'
-      @raw = Timeout.timeout(timeout) { `#{command}`.to_s }
+
+      UI.command(command) if print
+
+      tries = 1
+      begin
+        Timeout.timeout(timeout) do
+          result = `#{command}`.to_s # Using Helper.backticks doesn't work. Timeout don't times out and the command hangs forever
```

Comment:
Do you prefer the return here? In the new code I created the variable in an outside scope. The only reason I didn't return here was because of the printing the output, I didn't want that to be part of the `retry`.